### PR TITLE
[Analytics Hub] Create shared set of stats helpers for My Store and Analytics Hub

### LINKED ISSUE #8149: [Analytics Hub] Configure Orders Analytics card
Depends on #8148, #8156

## Summary
The orders analytics card should be able to display two types of information: `Total orders` and `Average Order Value`. Both can be properly acquired directly from the Woo Core stats API.

We will integrate real data from the API in #8189. Until then, we can use fake data for the previous and current period, and generate a percentage delta value based on the Total Orders and Average Order Value difference between the two periods.

## Designs

Image

### PR DESCRIPTION
**Pull request by rachelmcr**
Part of #8149

## Description

This PR moves a number of stats data helpers from the `StoreStatsPeriodViewModel` (used for My Store stats) to a separate `StatsV4DataHelper` class, so they can be reused in the Analytics Hub.

It also adds a new helper related to the new Average Order Value stat that will be used in Analytics Hub.

## Changes

* Moves the shared helpers and unit tests from `StoreStatsPeriodViewModel` to `StatsV4DataHelper`.
* Adds a new public helper `StatsV4DataHelper.createAverageOrderValueText(orderStats:currencyFormatter:currencyCode:)`.

## Testing

This PR refactors the helpers but there should be no change in the current behavior:

1. Build and run the app.
2. Confirm your stats (text and chart) appear as expected on the My Store screen.
3. Confirm you can select different time period or select specific intervals on the stats chart, and the stats text is updated as expected.

**Note:** This PR removes some redundant view model unit tests, because the logic is now contained in the helper class. However, you can undo the changes from 466c36f and run the tests in `StoreStatsPeriodViewModelTests` to confirm that all of the previous tests are still passing with these changes.

## Submitter Checklist

Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Dashboard/Factories/StatsV4DataHelper.swift [hunk 1]
```diff
@@ -0,0 +1,157 @@
+import Foundation
+import Yosemite
+import WooFoundation
+
+/// Helpers for calculating and displaying Stats V4 data
+///
+final class StatsV4DataHelper {
```
Comment (Ecarrion) [inline WooCommerce/Classes/ViewRelated/Dashboard/Factories/StatsV4DataHelper.swift]: This looks better suited for a `struct` type, but not a biggie!

Comment (Ecarrion) [reply WooCommerce/Classes/ViewRelated/Dashboard/Factories/StatsV4DataHelper.swift]: Also, how do you feel about naming it along the lines of `StatsDataTextFormatter`. 

Reason: Helper is quite a broad name, having a more concrete name limits the type of responsibility which makes it easy to maintain.

Also, Not a big deal 😅

**Review by Ecarrion — APPROVED** [review]
Looks good! I wonder if we should start calling this just `StatsDataHelper` and drop the `V4` since it seems like the standard by now!

File: WooCommerce/Classes/ViewRelated/Dashboard/Factories/StatsV4DataHelper.swift (hunk 1 above)
Comment (rachelmcr) [reply WooCommerce/Classes/ViewRelated/Dashboard/Factories/StatsV4DataHelper.swift]: Good points! I'll change it!

**Pull request merged**